shared/src/main/java/uk/gov/di/authentication/shared/lambda/BaseOrchestrationHandler.java

```diff
@@ -0,0 +1,98 @@
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.RequestHandler;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyResponseEvent;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.ThreadContext;
+import uk.gov.di.authentication.shared.entity.ClientSession;
+import uk.gov.di.authentication.shared.entity.ErrorResponse;
+import uk.gov.di.authentication.shared.helpers.LogLineHelper;
+import uk.gov.di.authentication.shared.services.ClientSessionService;
+import uk.gov.di.authentication.shared.services.ConfigurationService;
+import uk.gov.di.authentication.shared.services.SessionService;
+import uk.gov.di.authentication.shared.state.UserSession;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static uk.gov.di.authentication.shared.domain.RequestHeaders.CLIENT_SESSION_ID_HEADER;
+import static uk.gov.di.authentication.shared.helpers.ApiGatewayResponseHelper.generateApiGatewayProxyErrorResponse;
+import static uk.gov.di.authentication.shared.helpers.InstrumentationHelper.segmentedFunctionCall;
+import static uk.gov.di.authentication.shared.helpers.LogLineHelper.UNKNOWN;
+import static uk.gov.di.authentication.shared.helpers.LogLineHelper.attachLogFieldToLogs;
+import static uk.gov.di.authentication.shared.helpers.LogLineHelper.attachSessionIdToLogs;
+import static uk.gov.di.authentication.shared.helpers.RequestHeaderHelper.getHeaderValueFromHeaders;
+
+public abstract class BaseOrchestrationHandler
+        implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+    private static final Logger LOG = LogManager.getLogger(BaseFrontendHandler.class);
+    private static final String CLIENT_ID = "client_id";
+    protected final ConfigurationService configurationService;
+    protected final SessionService sessionService;
+    protected final ClientSessionService clientSessionService;
+
+    protected BaseOrchestrationHandler(
+            ConfigurationService configurationService,
+            SessionService sessionService,
+            ClientSessionService clientSessionService) {
+        this.configurationService = configurationService;
+        this.sessionService = sessionService;
+        this.clientSessionService = clientSessionService;
+    }
+
+    protected BaseOrchestrationHandler(ConfigurationService configurationService) {
+        this.configurationService = configurationService;
+        this.sessionService = new SessionService(configurationService);
+        this.clientSessionService = new ClientSessionService(configurationService);
+    }
+
+    @Override
+    public APIGatewayProxyResponseEvent handleRequest(
+            APIGatewayProxyRequestEvent input, Context context) {
+        return segmentedFunctionCall(
+                "ipv-api::" + getClass().getSimpleName(),
```

Comment:
Is this always the right segment?

Reply:
Yeah, this is one I was a little unsure of. I guess it's going to depend on where this is implemented, and will change when we move to our own repo. I'm not really sure what we should replace if with though